vine/src/parser.rs

```diff
@@ -248,7 +248,11 @@ impl<'ctx, 'src> VineParser<'ctx, 'src> {
 
   fn _parse_expr_prefix(&mut self, span: usize) -> Parse<'src, ExprKind> {
     if self.eat(Token::Return)? {
-      return Ok(ExprKind::Return(Box::new(self.parse_expr_bp(BP::ControlFlow)?)));
+      if self.eat(Token::Semi)? {
```

Comment:
```suggestion
      if self.check(Token::Semi) {
```

this will leave the `Token::Semi` to be consumed by `parse_stmt`; that way `(return; + 5)` is an error

Reply:
woops was meant to be check